### DISCUSSION TITLE
Add support for downloading Windows DLLs from Bitbucket

### DISCUSSION THREAD

**Discussion by mbabker** [php/pie#296] [Ideas]
"Promoting" https://github.com/php/pie/issues/110#issuecomment-2490606561 up to its own feature request as that thread seems more focused on Private Packagist (or other Packagist-like repos) than this one.

I was going to open a feature request for the [event extension](https://pecl.php.net/package/event) to add support for PIE, but as they use Bitbucket, they wouldn't be able to add Windows support at this time.

**Comment by asgrim** [comment]
Thanks @mbabker - I'll treat this as a specific ask for Bitbucket support; we can add this, I'd rather add support for specific repo providers if/when they come up, rather than just trying to support "everything" (e.g. lets say we add Gitlab support, but then no-one uses it, it's kinda waste of time heh).

**Comment by macintoshplus** [comment]
@mbabker your request is like mine. I mean Private Packagist for simplicity ;-)

**Comment by asgrim** [comment]
@mbabker sorry for the delay, I've been looking into Bitbucket, and it seems (unless I'm mistaken) they don't have the concept of a "release" like GitHub/GitLab where artifacts can be uploaded. However, there are general project downloads; for example https://bitbucket.org/asgrim/example-pie-repository/downloads/ could have the Windows DLL ZIP packages uploaded to it.

Just to note on this: GitHub does have [php/php-windows-builder](https://github.com/php/php-windows-builder) actions, which automatically builds and uploads. I don't believe there is anything similar for Bitbucket, so either someone would have to upload them manually in the correct expected format/naming.

That said, we can use Bitbucket API (https://developer.atlassian.com/cloud/bitbucket/rest/api-group-downloads/#api-group-downloads / https://developer.atlassian.com/cloud/bitbucket/rest/api-group-downloads/#api-repositories-workspace-repo-slug-downloads-filename-get) to fetch the downloads. Do you think that would work for the `event` extension?
